src/charm.py

```diff
@@ -572,6 +676,123 @@ def _add_cloud_integrator(
                 },
                 pipelines=["logs"],
             )
+        if tempo_url:
+            self.otel_config.add_exporter(
+                name="otlphttp/cloud-integrator",
+                exporter_config={
+                    "endpoint": tempo_url,
+                    "tls": {"insecure_skip_verify": insecure_skip_verify},
+                    **exporter_auth_config,
+                },
+                pipelines=["traces"],
+            )
+
+    def _get_tracing_receiver_url(self, protocol: ReceiverProtocol, tls_enabled: bool):
+        """Build the endpoint for the tracing receiver based on the protocol and TLS.
+
+        Args:
+            protocol: The ReceiverProtocol of a certain receiver (e.g., 'otlp_grpc', 'zipkin').
+            tls_enabled: Flag indicating whether the endpoint should use 'https' or not.
+        """
+        scheme = "http"
+        if tls_enabled:
+            scheme = "https"
+
+        # The correct transport protocol is specified in the tracing library, and it's always
+        # either http or grpc.
+        # We assume the user of the receiver is in-model, since this charm doesn't have ingress.
+        if receiver_protocol_to_transport_protocol[protocol] == TransportProtocolType.grpc:
+            return f"{socket.getfqdn()}:{PORTS.OTLP_GRPC}"
+        return f"{scheme}://{socket.getfqdn()}:{PORTS.OTLP_HTTP}"
+
+    @property
+    def _tracing_sampling(self) -> Dict[str, Any]:
```

Comment:
Currently, this is effectively a module-level constant.
Would it make sense to convert it to a free function?